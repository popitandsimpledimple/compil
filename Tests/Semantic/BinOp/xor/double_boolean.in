var a: double;
var b: boolean;

begin
    write(a xor b);
end.
