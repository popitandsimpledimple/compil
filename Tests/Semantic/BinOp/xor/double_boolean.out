program
   decls
      vars
         var
            a
            type
               double
            no expression
      vars
         var
            b
            type
               boolean
            no expression
   block
      compounds statements
         call
            write
               xor
                  a
                  b

 ───────────────────────────────────────────────────────────────────────────────────────────────────
 │ integer                   │ type                      │ integer                                 │
 │ char                      │ type                      │ char                                    │
 │ string                    │ type                      │ string                                  │
 │ double                    │ type                      │ double                                  │
 │ boolean                   │ type                      │ boolean                                 │
 ───────────────────────────────────────────────────────────────────────────────────────────────────

 ───────────────────────────────────────────────────────────────────────────────────────────────────
 │ without name              │ program name              │ no return type                          │
 │ a                         │ var                       │ double                                  │
 │ b                         │ var                       │ boolean                                 │
 ───────────────────────────────────────────────────────────────────────────────────────────────────
